GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift

```diff
@@ -181,21 +181,37 @@ final class AreaBarChartView : UIView, ViewModelRendering {
             view.removeFromSuperview()
         }
 
+        var viewsToPutToFront = [UIView]()
+
         viewModel.mainChart.barItems.forEach { itemViewModel in
             let barView = barStackView.addBarWithItemViewModel(itemViewModel)
-            addLineWithLabelToBarView(barView, withViewModel: itemViewModel)
+            let lineView = addLineWithLabelToBarView(barView, withViewModel: itemViewModel)
+
+            switch itemViewModel.valueLabelMode {
+            case .VisibleWithHiddenLabel, .Hidden:
+                break
+            default:
+                viewsToPutToFront.append(lineView)
+            }
+
+        }
+
+        viewsToPutToFront.forEach {
+            bringSubviewToFront($0)
```

Comment:
I am not sure if we need `viewsToPutToFront` array. Only `lineView`s are stored there, and `barStackView` and `informationSheet` are passed to `bringSubviewToFront` separately, few lines later.